service/awsconfig/v5/resource/cloudformation/adapter/outputs.go

```diff
@@ -1,20 +1,70 @@
 package adapter
 
-import "github.com/giantswarm/aws-operator/service/awsconfig/v5/cloudconfig"
+import (
+	"strconv"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/cloudconfig"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v5/key"
+)
 
 // The template related to this adapter can be found in the following import.
 //
 //     github.com/giantswarm/aws-operator/service/awsconfig/v5/templates/cloudformation/guest/outputs.go
 //
 
 type outputsAdapter struct {
-	MasterCloudConfigVersion string
-	WorkerCloudConfigVersion string
+	Master        outputsAdapterMaster
+	Worker        outputsAdapterWorker
+	VersionBundle outputsAdapterVersionBundle
+}
+
+type outputsAdapterMaster struct {
+	ImageID      string
+	InstanceType string
+	CloudConfig  outputsAdapterMasterCloudConfig
+}
+
+type outputsAdapterMasterCloudConfig struct {
+	Version string
+}
+
+type outputsAdapterWorker struct {
+	Count        string
+	ImageID      string
+	InstanceType string
+	CloudConfig  outputsAdapterWorkerCloudConfig
 }
 
-func (o *outputsAdapter) getOutputs(cfg Config) error {
-	o.MasterCloudConfigVersion = cloudconfig.MasterCloudConfigVersion
-	o.WorkerCloudConfigVersion = cloudconfig.WorkerCloudConfigVersion
+type outputsAdapterWorkerCloudConfig struct {
+	Version string
+}
+
+type outputsAdapterVersionBundle struct {
+	Version string
+}
+
+func (a *outputsAdapter) Adapt(config Config) error {
+	imageID, err := key.ImageID(config.CustomObject)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	workerCount := key.WorkerCount(config.CustomObject)
```

Comment:
There's a lot of duplication here, these values need to be available for other adapters too.